include/openthread/platform/radio.h

```diff
@@ -88,7 +88,11 @@ enum
     OT_RADIO_CHANNEL_PAGE_0_MASK = (1U << OT_RADIO_CHANNEL_PAGE_0), ///< 2.4 GHz IEEE 802.15.4-2006
     OT_RADIO_CHANNEL_PAGE_2      = 2,                               ///< 915 MHz IEEE 802.15.4-2006
     OT_RADIO_CHANNEL_PAGE_2_MASK = (1U << OT_RADIO_CHANNEL_PAGE_2), ///< 915 MHz IEEE 802.15.4-2006
-    OT_RADIO_CHANNEL_PAGE_MAX    = OT_RADIO_CHANNEL_PAGE_2,         ///< Maximum supported channel page value
+#if OPENTHREAD_CONFIG_PLATFORM_RADIO_PROPRIETARY_SUPPORT
+    OT_RADIO_CHANNEL_PAGE_MAX = 0xFF,
```

Comment:
Is setting to 255 problematic for this code:
https://github.com/openthread/openthread/blob/8b63efe2aed519647c8173b5317b5331b8ba59ff/src/core/radio/radio.hpp#L105
?

I wonder if we should just remove the `OT_RADIO_CHANNEL_PAGE_MAX`. I'm not sure what value it provides.

Reply:
Hi @jwhui , I used 255 as the spec currently provides an entire byte for this field.  However, I agree that setting the channel page to any value greater than 31 is going to be problematic for `kSupportedChannelPages`. While one option could be to remove OT_RADIO_CHANNEL_PAGE_MAX, I am wondering if we should instead keep it set to 31 (instead of 255) for the proprietary case. Thoughts?

Reply:
A third option could be - to remove OT_RADIO_CHANNEL_PAGE_MAX, and add a compile-time check to make sure that the proprietary channel page value (set by platform) is not greater than 31.

Reply:
> A third option could be - to remove OT_RADIO_CHANNEL_PAGE_MAX, and add a compile-time check to make sure that the proprietary channel page value (set by platform) is not greater than 31.

Let's go with this option, if that's OK with you.

Reply:
Done. Thanks.